src/context/product-context.js

```diff
@@ -0,0 +1,51 @@
+import {
+  createContext,
+  useContext,
+  useReducer,
+  useState,
+  useEffect,
+} from "react";
+import useAxios from "../utils/useAxios";
+import filterReducerFunc from "../reducer/filterReducerFunc";
+import {
+  sortByHighToLowFunc,
+  sortByLowToHighFunc,
+  ratingFunc,
+  categoryFunc,
+} from "../utils/filterFunc";
+const ProductContext = createContext();
+const compose = (state, ...allFilters) => (data) => {
+  return allFilters.reduce((acc, curr) => {
+    return curr(state, acc);
+  }, data);
+};
+const ProductContextProdvider = ({ children }) => {
+  const [prodList, setProdList] = useState([]);
+  const { apiResponse, loading } = useAxios("/api/products");
+  console.log(apiResponse.products);
```

Comment:
Remove the console.logs if done with testing out.

Reply:
Thank you for pointing out Removed Logs